### PR TITLE
Feat/contract input generation ui

### PR DESCRIPTION
**Pull request by azara**
Feat/contract input generation ui

### CODE REVIEW

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/FieldToContractInputMappingViewerCheckStateManager.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+     */
+    @Override
+    public void checkStateChanged(final CheckStateChangedEvent event) {
+        final FieldToContractInputMapping mapping = (FieldToContractInputMapping) event.getElement();
+        mapping.setGenerated(event.getChecked());
+        final CheckboxTreeViewer checkboxTreeViewer = (CheckboxTreeViewer) event.getSource();
+        checkboxTreeViewer.setSubtreeChecked(mapping, event.getChecked());
+        selectParentIfChildIsSelected(event, mapping, checkboxTreeViewer);
+        deselectParentIfNoChildSelected(event, mapping, checkboxTreeViewer);
+        // checkboxTreeViewer.setGrayChecked(mapping, isGrayed(mapping));
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/FieldToContractInputMappingViewerCheckStateManager.java]: commented code and extra empty line

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/FieldToContractInputMappingViewerCheckStateManager.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    /**
+     * @param event
+     * @param mapping
+     * @param checkboxTreeViewer
+     */
+    private void deselectParentIfNoChildSelected(final CheckStateChangedEvent event, final FieldToContractInputMapping mapping,
+            final CheckboxTreeViewer checkboxTreeViewer) {
+        if (!event.getChecked() && mapping.getParent() != null) {
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/FieldToContractInputMappingViewerCheckStateManager.java]: extract mapping.getPrent in a variable as it is called 3 times

File: bundles/plugins/org.bonitasoft.studio.contract/src-test/java/org/bonitasoft/studio/contract/ui/wizard/FieldToContractInputMappingObservableFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
+package org.bonitasoft.studio.contract.ui.wizard;
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src-test/java/org/bonitasoft/studio/contract/ui/wizard/FieldToContractInputMappingObservableFactoryTest.java]: missing header

File: bundles/plugins/org.bonitasoft.studio.contract/src-test/java/org/bonitasoft/studio/contract/ui/wizard/labelProvider/FieldTypeColumnLabelProviderTest.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package org.bonitasoft.studio.contract.ui.wizard.labelProvider;
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src-test/java/org/bonitasoft/studio/contract/ui/wizard/labelProvider/FieldTypeColumnLabelProviderTest.java]: Missing header

File: bundles/plugins/org.bonitasoft.studio.contract/src-test/java/org/bonitasoft/studio/contract/ui/wizard/labelProvider/InputTypeColumnLabelProviderTest.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
+package org.bonitasoft.studio.contract.ui.wizard.labelProvider;
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src-test/java/org/bonitasoft/studio/contract/ui/wizard/labelProvider/InputTypeColumnLabelProviderTest.java]: missing header

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/property/ContractPropertySection.java [hunk 6]
```diff
@@ -194,6 +205,34 @@ private void createInputTabContent(final Composite parent, final IObservableValu
...
+             * (non-Javadoc)
+             * @see org.eclipse.swt.events.SelectionAdapter#widgetSelected(org.eclipse.swt.events.SelectionEvent)
+             */
+            @Override
+            public void widgetSelected(final SelectionEvent e) {
+                openAddInputWizardDialog();
+            }
+
+            /**
+             *
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/property/ContractPropertySection.java]: remove this empty javadoc

**Comment by cmckinty** [comment]
I made 2 comments: wich -> which (in two places)

Also commented on a code comment... not top priority!

Chris


On 7 May 2015 at 10:46, azara <notifications@github.com> wrote:

> Assigned #65 <https://github.com/bonitasoft/bonita-studio/pull/65> to
> @cmckinty <https://github.com/cmckinty>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bonitasoft/bonita-studio/pull/65#event-299039173>.
>



-- 
*Christine McKinty*
*Technical Writer*
+33615650047Download Bonita BPM
<http://www.bonitasoft.com/landing/lp/direct-download?utm_source=bonitasoft_email_signature&utm_medium=email&utm_content=email_signature&utm_campaign=email_signature>
 | Access BPM Quick Start Center
<http://www.bonitasoft.com/landing/lp/quick-start-center?utm_source=bonitasoft_email_signature&utm_medium=email&utm_content=email_signature&utm_campaign=email_signature>
   <http://www.bonitasoft.com/> <http://twitter.com/bonitasoft>
<https://www.facebook.com/bonitasoftbpm>
<http://www.linkedin.com/groups?gid=3182097>
<http://www.slideshare.net/BonitaSoft> <http://www.youtube.com/bonitasoft>
<http://www.bonitasoft.com/bos_redirect.php?bos_redirect_id=128> This
message and
any attachment (the "message") is intended solely for the addressees and is
confidential. If you receive this message by mistake, please delete it and
notify the sender immediately. Any use not in accordance with its purpose,
any out-spread or disclosure, either as a whole or partially, is prohibited
except with formal approval. Internet cannot guarantee the integrity of
this message, therefore Bonitasoft will not be liable for the message if
modified.

**Pull request merged**